geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientRegistrationMetadata.java

```diff
@@ -47,51 +49,67 @@
   private KnownVersion clientVersion;
   private DataInputStream dataInputStream;
   private DataOutputStream dataOutputStream;
+  private final NioFilter ioFilter;
 
-  ClientRegistrationMetadata(final InternalCache cache, final Socket socket) {
+  ClientRegistrationMetadata(final InternalCache cache, final Socket socket,
+      final NioFilter ioFilter) {
     this.cache = cache;
     this.socket = socket;
     socketMessageWriter = new SocketMessageWriter();
+    this.ioFilter = ioFilter;
   }
 
   boolean initialize() throws IOException {
-    DataInputStream unversionedDataInputStream = new DataInputStream(socket.getInputStream());
-    DataOutputStream unversionedDataOutputStream = new DataOutputStream(socket.getOutputStream());
-
-    if (getAndValidateClientVersion(socket, unversionedDataInputStream,
-        unversionedDataOutputStream)) {
-      if (oldClientRequiresVersionedStreams(clientVersion)) {
-        dataInputStream =
-            new VersionedDataInputStream(unversionedDataInputStream, clientVersion);
-        dataOutputStream =
-            new VersionedDataOutputStream(unversionedDataOutputStream, clientVersion);
+    InputStream inputStream = null;
+    try {
+      if (ioFilter == null) {
+        inputStream = socket.getInputStream();
       } else {
-        dataInputStream = unversionedDataInputStream;
-        dataOutputStream = unversionedDataOutputStream;
+        inputStream = ioFilter.getInputStream(socket);
```

Comment:
shouldn't the initialization of inputStream happen right before the try (instead of after) since all the finally block does is close the inputStream? Also if you then extracted these lines of code into a method you could then have: final InputStream inputStream = getInputStream(socket, ioFilter); try ...

Reply:
Updated